libminifi/src/capi/api.cpp

```diff
@@ -170,31 +165,31 @@ flow_file_record* create_ff_object_na(const char *file, const size_t len, const
   new_ff->attributes = nullptr;
   new_ff->contentLocation = new char[len + 1];
   snprintf(new_ff->contentLocation, len + 1, "%s", file);
-  std::ifstream in(file, std::ifstream::ate | std::ifstream::binary);
   // set the size of the flow file.
   new_ff->size = size;
+  new_ff->crp = static_cast<void*>(new std::shared_ptr<minifi::core::ContentRepository>);
```

Comment:
make_shared ?

Reply:
As there are no arguments provided, I couldn't save specifying the template parameter. Moreover it's an empty shared ptr, so it doesn't even allocate twice.